tket2/src/rewrite/trace.rs

```diff
@@ -0,0 +1,126 @@
+//! Utilities for tracing the rewrites applied to a circuit.
+//!
+//! This is only tracked if the `rewrite-tracing` feature is enabled.
+
+use hugr::hugr::hugrmut::HugrMut;
+use hugr::hugr::NodeMetadata;
+use itertools::Itertools;
+
+use crate::Circuit;
+
+use super::CircuitRewrite;
+
+/// Metadata key for the circuit rewrite trace.
+pub const METADATA_REWRITES: &str = "TKET2.rewrites";
+
+/// Global read-only flag for enabling rewrite tracing.
+/// Enable it by setting the `rewrite-tracing` feature.
+///
+/// Note that circuits must be explicitly enabled for rewrite tracing by calling
+/// [`RewriteTracer::enable_rewrite_tracing`].
+pub const REWRITE_TRACING_ENABLED: bool = cfg!(feature = "rewrite-tracing");
```

Comment:
is rust analyzer etc. good enough to ignore code guarded by this constant?

Reply:
rust analyser checks the default features unless configured differently, bit this should work as any const flag.